src/Materials/multiMaterial.ts

```diff
@@ -245,21 +247,32 @@ export class MultiMaterial extends Material {
         var multiMaterial = new MultiMaterial(parsedMultiMaterial.name, scene);
 
         multiMaterial.id = parsedMultiMaterial.id;
+        multiMaterial._loadedUniqueId = parsedMultiMaterial.uniqueId;
 
         if (Tags) {
             Tags.AddTagsTo(multiMaterial, parsedMultiMaterial.tags);
         }
 
-        for (var matIndex = 0; matIndex < parsedMultiMaterial.materials.length; matIndex++) {
-            var subMatId = parsedMultiMaterial.materials[matIndex];
-
-            if (subMatId) {
-                // If the same multimaterial is loaded twice, the 2nd multimaterial needs to reference the latest material by that id which
-                // is why this lookup should use getLastMaterialByID instead of getMaterialByID
-                multiMaterial.subMaterials.push(scene.getLastMaterialById(subMatId));
-            } else {
-                multiMaterial.subMaterials.push(null);
-            }
+        if (parsedMultiMaterial.materialsUniqueIds) {

```

Comment:
Again I wonder if we can refactor that code?

Reply:
I can make it a function that we call twice but I am not sure that will improve readability

Reply:
I feel bad when I see the same line multiple times ;D

Reply:
Oh, I found a way. How's that?

Reply:
great